common/src/main/java/org/wso2/carbon/testgrid/common/TestPlan.java

```diff
@@ -58,9 +59,23 @@
     private Script deploymentScript;
 
     public enum Status {
-        EXECUTION_PLANNED, INFRASTRUCTURE_PREPARATION, INFRASTRUCTURE_READY, INFRASTRUCTURE_ERROR, DEPLOYMENT_PREPARATION,
-        DEPLOYMENT_READY, DEPLOYMENT_ERROR, SCENARIO_EXECUTION, SCENARIO_EXECUTION_ERROR, SCENARIO_EXECUTION_COMPLETED,
-        REPORT_GENERATION, REPORT_GENERATION_ERROR, EXECUTION_COMPLETED
+        EXECUTION_PLANNED ("Execution Planned"), INFRASTRUCTURE_PREPARATION ("Infrastructure Preparation"),
+        INFRASTRUCTURE_READY ("Infrastructure Ready"), INFRASTRUCTURE_ERROR ("Infrastructure Error"),
+        INFRASTRUCTURE_DESTROY_ERROR ("Infrastructure Destroy Error"), DEPLOYMENT_PREPARATION ("Deployment Preparation"),
+        DEPLOYMENT_READY ("Deployment Ready"), DEPLOYMENT_ERROR ("Deployment Error"),
+        SCENARIO_EXECUTION ("Scenario Execution"), SCENARIO_EXECUTION_ERROR ("Scenario Execution Error"),
+        SCENARIO_EXECUTION_COMPLETED ("Scenario Execution Completed"), REPORT_GENERATION ("Report Generation"),
+        REPORT_GENERATION_ERROR ("Report Generation Error"), EXECUTION_COMPLETED ("Execution Completed");
+
+        private final String name;
+
+        Status(String s) {
```

Comment:
Shall we rename `s` to `status since it makes more sense? WDYT?

Reply:
Fixed with f243d73cb375a67c4fb173c6c2d75a931d9411ae.